### ISSUE TITLE
Refine `function wordToFiles( word )`

### ISSUE THREAD

**Issue by hugolpz** [lingua-libre/SignIt#51]
Current intersection from clicked-word via `window.select()` and list of videos files is handled by [`function wordToFiles( word )`](https://github.com/lingua-libre/SignIt/blob/bd88c67aaccba9ec451435ca249b21b5afda02c8/background-script.js#L275-L281).

This function is too restrictive.

### Additional cases to handle
| Selected word | existing Wikimedia commons file | string
|:---|:---|:--- 
|`accueil` | [`accueil (lieu)`](https://commons.wikimedia.org/wiki/File:LL-Q33302_(fsl)-Hugo_en_résidence-accueil_(lieu).webm) | File:LL-Q33302_(fsl)-Hugo_en_résidence-accueil_(lieu).webm

### Participative listing
Other cases are observed to not work.

**Comment by pushkar707** [comment]
I have solved this to some extent.

I have put a check to find if there's a single word from the complete selection whose record is available.

If there is, we return a record for the first word whose record is there.

if this is okay, I will make a PR.

**Comment by hugolpz** [comment]
@pushkar707 , this is not okay, this is **great** Q___Q

Your PR is welcome yes ! 🌻

**Comment by pushkar707** [comment]
Thanks a lot :), I have already made PR #57.

I was not able to link it to the issue. Please check it.

**Comment by hugolpz** [comment]
@pushkar707 I never though this far to be honest (word carousel). I took over this extension in a lower point it took be some time to bring it up to current status.

What I can confirm is we have videos names with the convention of `word (clarification)`, such as :
- `hi (hello).webm`
- `hi (surprise).webm`
- `hi (scream).webm`

When double clicking `hi` in `**hi** how are you ?`, the 3 videos above should be available on the video gallery.

**Comment by kabir-afk** [comment]
> @pushkar707 I never though this far to be honest. I took over this extension in a lower point it took be some time to bring it up to current status.
> 
> ### Convention
> What I can confirm is we have videos names with the convention of `word (clarification)`, such as :
> 
> * `hi (hello).webm`
> * `hi (surprise).webm`
> * `hi (scream).webm`
> 
> Note: our videos are French Sign Language for now due to lack of contributors from other countries but the principle is true.
> 
> ### What we want
> Given text such as:
> 
> > **hi** how are you ?
> 
> When double clicking `hi`, the 3 videos cited above should be available on the video gallery.
> 
> I'm not sure the split by space is the best, but it implements a split and collection of words from the split, so it's interesting. 👍🏼 We can build upon your code, split by `(`, and keep `parts[0]` if we decide to.

You are right, while approach taken by @pushkar707 is ideal , it needs a lil bit of refactoring as it dies down when punctuation marks come in between. For example it'll work for `croissant dépassant`. But if something like `croissant, dépassant`, i.e., a `croissant` with a comma , it doesn't work as there is no `croissant,` inside `records` . I'll try to refactor it.

**Comment by hugolpz** [comment]
@kabir-afk , regex improvement discussions come here.

**Comment by hugolpz** [comment]
Hello @kabir-afk ,
I have difficulties to understand the current regex. But simply put, if it split by punctuation except `-`, then we can close this issue for now.

See also :
- [regex code](https://github.com/lingua-libre/SignIt/blob/master/content_scripts/signit.js#L109-L110)

**Comment by kabir-afk** [comment]
Naah , I don't think so. The issue states that sometimes even if the files exist ( as indicated by `colorThoseWords` function ), SignIt is unable to find it. This shows that function is working as expected and like you mentioned before , `sparqlSignVideosQuery` needs to be worked upon in order to address the issue better.

As far as the above regex is concerned, you are right that it checks for punctuations/non-word characters and then later applies `global` and `insensitive case` flags inside the constructor function. We could have also used `/w` or something like that , but that would have ignored the accented letters as well, which shouldn't be the case , hence the big and complex regex.
